src/ring_theory/ideal/operations.lean

```diff
@@ -1604,3 +1604,105 @@ begin
+  quot_left_to_quot_sum I J x = 0 :=
+begin
+  have hIJmap: ((quot_left_to_quot_sum I J).comp(ideal.quotient.mk I) '' J) =
+    (ideal.quotient.mk (I+J) '' J),
+   {apply set.ext,
+    intro y,
+    split,
+
+     {intro hy,
+      obtain ⟨z,hz⟩ := (set.mem_image ((quot_left_to_quot_sum I J).comp(ideal.quotient.mk I))
+        J y).1 hy,
+      unfold quot_left_to_quot_sum at hz,
+      rw [ring_hom.comp_apply,ideal.quotient.lift_mk] at hz,
+      rw ← hz.right,
+      exact set.mem_image_of_mem (ideal.quotient.mk (I+J)) hz.left},
+
+     {intro hy,
+      obtain ⟨z,hz⟩ := (set.mem_image (ideal.quotient.mk (I+J)) J y).1 hy,
+      rw [quot_left_to_quot_sum, set.mem_image_eq],
+      use z,
+      rwa [ring_hom.comp_apply, ideal.quotient.lift_mk]},
+    },
+
+  have hJ: (J.map (ideal.quotient.mk I)).map (quot_left_to_quot_sum I J) = J.map
+    (ideal.quotient.mk (I+J))
+    := by rw [ideal.map_map, ideal.map,hIJmap, ← ideal.map],
+
+  have hmapx : quot_left_to_quot_sum I J x ∈ (J.map (ideal.quotient.mk I)).map
+    (quot_left_to_quot_sum I J),
+    {rw ideal.map,
+    apply set.mem_of_subset_of_mem (ideal.subset_span),
+    exact (set.mem_image_of_mem (quot_left_to_quot_sum I J) hx)},
+
+  rwa [hJ, in_ker_proj_to_add_right I J] at hmapx,
+end
+
+/-- define `double_quot_to_quot_add` to be the induced ring hom `(R/I)/J' ->R/(I+J)`,
+  where `J'` is the image of `J` in `R/I` -/
+def double_quot_to_quot_add : (J.map (ideal.quotient.mk I)).quotient →+* (I + J).quotient :=
+ideal.quotient.lift (ideal.map (ideal.quotient.mk I) J) (quot_left_to_quot_sum I J)
+ (img_left_in_ker I J)
+
+/-- define `double_quot_mk` to be the composite of the maps `R → (R/I) and (R/I) → (R/I)/J'` -/
+def double_quot_mk := ring_hom.comp (ideal.quotient.mk (J.map (ideal.quotient.mk I)))
+ (ideal.quotient.mk I)
+
+-- Another short result for lifting map `ring_to_double_quot` to a map `R/(I+J) → (R/I)/J'`
+--shorter and easier proof using ker_g₁ from
+def mem_add_double_quot_mk (x : R) (hx : x ∈ I+J) : double_quot_mk I J x = 0 :=
+begin
+  have hIJtoJ : (I+J).map(ideal.quotient.mk I) = J.map(ideal.quotient.mk I) := by {
+    rw [ideal.add_eq_sup, ideal.map_sup, ideal.map_quotient_self],
+    simp},
+  have : ((I+J).map(ideal.quotient.mk I)).map(ideal.quotient.mk
+    (J.map(ideal.quotient.mk I))) = ⊥ := by rw [hIJtoJ, ideal.map_quotient_self
+      (J.map(ideal.quotient.mk I))],
+  rw [double_quot_mk, ← ideal.mem_bot, ← this, ring_hom.comp_apply],
+  apply ideal.mem_map_of_mem,
+  apply ideal.mem_map_of_mem,
+  exact hx,
+end
+
+-- define `lift_add_double_qot_mk` to be the induced map `R/(I+J) → (R/I)/J' `
+def lift_add_double_qot_mk (I J : ideal R) := ideal.quotient.lift (I+J) (double_quot_mk I J)
+  (mem_add_double_quot_mk I J)
+
+-- Then `double_quot_to_quot_add` and `lift_add_double_qot_mk` are inverse isomorphisms
+theorem double_quot_equiv_quot_add : (J.map (ideal.quotient.mk I)).quotient ≃+* (I + J).quotient :=
```

Comment:
If you run `#lint` at the bottom of the file (or wait for CI to run it), it will tell you that this should be a def not a lemma.